src/portals/hope/features/dashboard/TasksData.tsx

```diff
@@ -107,16 +114,39 @@ interface TasksDataProps {
 type AreaOrNull = TaskResourceArea | null
 
 export const TasksData = ({ tasks }: TasksDataProps) => {
-  const [selectedAreas, setSelectedAreas] = useState<AreaOrNull[]>([])
+  const [selectedAreas, setSelectedAreas] = useLocalStorage<AreaOrNull[]>(
+    'DashboardSelectedAreas',
+    [],
+  )
+  const [selectedMarkets, setselectedMarkets] = useLocalStorage<string[]>(
+    'DashboardSelectedMarkets',
+    [],
+  )
+
+  const filteredTasksByMarket =
+    selectedMarkets.length === 0
+      ? tasks
+      : tasks.filter(
+          (task) => task.market && selectedMarkets.includes(task.market),
+        )
 
   const filteredTasks =
     selectedAreas.length === 0
-      ? tasks
-      : tasks.filter((task) => selectedAreas.includes(task.area ?? null))
+      ? filteredTasksByMarket
+      : filteredTasksByMarket.filter(
+          (task) => task.market && selectedAreas.includes(task.area ?? null),
```

Comment:
```suggestion
          (task) => selectedAreas.includes(task.area ?? null),
```
since we can also have `null` as area 🫠